tests/chainer_tests/links_tests/connection_tests/test_tree_lstm.py

```diff
@@ -33,19 +33,29 @@ def _child_sum_tree_lstm(func, *inputs):
         W_ha, W_hi, W_ho = xp.split(W_h_aio, 3, 1)
         W_hf = W_h_f
 
-        sum_h = sum(hs)
-        a = x.dot(W_xa) + sum_h.dot(W_ha) + b_a
-        i = x.dot(W_xi) + sum_h.dot(W_hi) + b_i
-        o = x.dot(W_xo) + sum_h.dot(W_ho) + b_o
-        f_list = [x.dot(W_xf) + h.dot(W_hf) + b_f for h in hs]
-
-        a = xp.tanh(a)
-        i = _sigmoid(i)
-        o = _sigmoid(o)
-        f_list = [_sigmoid(f) for f in f_list]
-
-        c_next = a * i + sum(f * c for f, c in zip(f_list, cs))
-        y = o * xp.tanh(c_next)
+        if len(hs) >= 1:
+            sum_h = sum(hs)
+            a = x.dot(W_xa) + sum_h.dot(W_ha) + b_a
+            i = x.dot(W_xi) + sum_h.dot(W_hi) + b_i
+            o = x.dot(W_xo) + sum_h.dot(W_ho) + b_o
+            f_list = [x.dot(W_xf) + h.dot(W_hf) + b_f for h in hs]
+
+            a = xp.tanh(a)
+            i = _sigmoid(i)
+            o = _sigmoid(o)
+            f_list = [_sigmoid(f) for f in f_list]
+
+            c_next = a * i + sum(f * c for f, c in zip(f_list, cs))
```

Comment:
How about `c_next = sum(f * c for f, c in zip(f_list, cs), a * i)`?
And, Please merge LL43-49 and LL54-58.

Reply:
Cool. I'll do so.

Reply:
Cool. I'll do that.

Reply:
The cool sum cause an error `SyntaxError: Generator expression must be parenthesized if not sole argument`.
So, I'll write `sum([f * c for f, c in zip(f_list, cs)], a * i)` instead.